etsin_finder/frontend/js/stores/view/qvain.js

```diff
@@ -74,6 +79,207 @@ export class Qvain {
   get addedParticipants() {
     return this.participants
   }
+
```

Comment:
Entire bunch of file picker functionality. Observables are named with underscore (_) in their name, so that the get functions can share the name (to make code prettier and more readable). Components should not directly interact with the observables, instead use functions to change the values.

Important note: modifying internal structures in a hierarchical object _seems_ to not make `mobx` to update subscribed components. Reassignment is the only assured way of making the subscribed components to update.